File: documentapi/src/main/java/com/yahoo/documentapi/messagebus/protocol/SearchResultMessage.java
// Copyright Yahoo. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
package com.yahoo.documentapi.messagebus.protocol;

import com.yahoo.vdslib.SearchResult;

public class SearchResultMessage extends VisitorMessage {

    private SearchResult searchResult = null;

    public SearchResult getResult() {
        return searchResult;
    }

    public void setSearchResult(SearchResult result) {
        searchResult = result;
    }

    @Override
    public DocumentReply createReply() {
        return new VisitorReply(DocumentProtocol.REPLY_SEARCHRESULT);
    }

    @Override
    public int getType() {
        return DocumentProtocol.MESSAGE_SEARCHRESULT;
    }
}

File: documentapi/src/vespa/documentapi/messagebus/messages/searchresultmessage.cpp
// Copyright Yahoo. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.

#include "searchresultmessage.h"

using vdslib::SearchResult;

namespace documentapi {

SearchResultMessage::SearchResultMessage() :
    VisitorMessage(),
    SearchResult()
{
    // empty
}

SearchResultMessage::SearchResultMessage(SearchResult &&result) :
    VisitorMessage(),
    SearchResult(std::move(result))
{
    // empty
}

DocumentReply::UP
SearchResultMessage::doCreateReply() const
{
    return std::make_unique<VisitorReply>(DocumentProtocol::REPLY_SEARCHRESULT);
}

uint32_t
SearchResultMessage::getApproxSize() const
{
    return SearchResult::getSerializedSize();
}

uint32_t
SearchResultMessage::getType() const
{
    return DocumentProtocol::MESSAGE_SEARCHRESULT;
}

}

